internal/controllers/common.go

```diff
@@ -17,3 +26,39 @@ func ImageIDToRBDID(imageID string) string {
 func SnapshotIDToRBDID(snapshotID string) string {
 	return SnapshotRBDIDPrefix + snapshotID
 }
+
+func getSnapshotSourceDetails(snapshot *providerapi.Snapshot) (string, string, error) {
+	var parentName, snapName string
+	switch {
+	case snapshot.Source.IronCoreImage != "":
+		parentName = SnapshotIDToRBDID(snapshot.ID)
+		snapName = ImageSnapshotVersion
+	case snapshot.Source.VolumeImageID != "":
+		parentName = ImageIDToRBDID(snapshot.Source.VolumeImageID)
+		snapName = snapshot.ID
+	default:
+		return "", "", fmt.Errorf("snapshot source is not present")
+	}
+	return parentName, snapName, nil
+}
+
+func listRbdImageChildren(ioCtx *rados.IOContext, imageID string) (int, int, error) {
```

Comment:
As mentioned in my other [comment](https://github.com/ironcore-dev/ceph-provider/pull/793/files#r2447774066), we don't need one of the `listRbdImageChildren` invocations in the image_controller. As only one invocation remains, I think we can delete this function and have the code inline.

This should avoid some of the error handling code here.